spyder_line_profiler/spyder/widgets.py

```diff
@@ -257,17 +345,25 @@ def analyze(self, filename=None, wdir=None, args=None, pythonpath=None,
 
     def select_file(self):
         self.redirect_stdio.emit(False)
+        try:    
+            curr_script = self.editor.get_current_filename()
+        except:
+            curr_script = None
+            
+        pwd = curr_script if curr_script else getcwd_or_home()
+        
         filename, _selfilter = getopenfilename(
-            self, _("Select Python script"), getcwd(),
+            self, _("Select Python script"), pwd,
             _("Python scripts")+" (*.py ; *.pyw)")
         self.redirect_stdio.emit(False)
+        
         if filename:
             self.analyze(filename)
 
     def show_log(self):
         if self.output:
-            editor = TextEditor(self.output, title=_("Line profiler output"),
-                                readonly=True)
+            editor = TextEditor(self.output, title=_("Line profiler output"))
```

Comment:
I think we are missing the `parent=self` kwarg when creating the `TextEditor` instance since right now is showing as if Spyder was using the clear theme:

![imagen](https://user-images.githubusercontent.com/16781833/162042520-43e11ddc-e98f-4bd5-a5b9-fe75f558f61d.png)

We also should add it in the `TextEditor` creation at the `show_errorlog` dialog

Also, I think `readonly=True` kwarg should still be passed since the `save and close` button doesn't do anything (as far as I tested) or maybe is there some reason why the button should be shown?